var/spack/repos/builtin/packages/gasnet/package.py

```diff
@@ -6,98 +6,133 @@
+    # GASNet-EX releases over 2 years old are not supported.
 
-    def url_for_version(self, version):
-        url = "https://gasnet.lbl.gov/"
-        if version >= Version('2019'):
-            url += "EX/GASNet-{0}.tar.gz".format(version)
-        else:
-            url += "download/GASNet-{0}.tar.gz".format(version)
-
-        return url
-
-    def configure_args(self):
-        args = [
-            # TODO: factor IB suport out into architecture description.
-            '--enable-par',
-            '--enable-mpi-compat',
-            '--enable-segment-fast',
-            '--disable-parsync',
-            '--with-segment-mmap-max=%s '
-            % (self.spec.variants['segment-mmap-max'].value),
-            # for consumers with shared libs
-            "CC=%s %s" % (spack_cc, self.compiler.cc_pic_flag),
-            "CXX=%s %s" % (spack_cxx, self.compiler.cxx_pic_flag),
-        ]
-
-        if '+aligned-segments' in self.spec:
-            args.append('--enable-aligned-segments')
-        else:
-            args.append('--disable-aligned-segments')
+    # The optional network backends:
+    variant('smp', default=True, description="SMP conduit for single-node operation")
+    variant('mpi', default=False, description="Low-performance/portable MPI conduit")
+    variant('ibv', default=False, description="Native InfiniBand verbs conduit")
+    variant('udp', default=False, description="Portable UDP conduit, for Ethernet networks")
+
+    variant('debug', default=False, description="Enable library debugging mode")
+
+    depends_on('mpi', when='+mpi')
 
-        if '+pshm' in self.spec:
-            args.append('--enable-pshm')
+    depends_on('autoconf@2.69', type='build', when='@master:')
+    depends_on('automake@1.16:', type='build', when='@master:')
+
+    def conduits(self):
+        return [c for c in ['smp', 'mpi', 'ibv', 'udp'] if "+" + c in self.spec]
+
+    def install(self, spec, prefix):
+        if spec.satisfies('@master:'):
+            bootstrapsh = Executable("./Bootstrap")
+            bootstrapsh()
+
+        # The GASNet-EX library has a highly multi-dimensional configure space,
+        # to accomodate the varying behavioral requirements of each client runtime.
+        # The library's ABI/link compatibility is strongly dependent on these
+        # client-specific build-time settings, and that variability is deliberately NOT
+        # encoded in the variants of this package. The recommended way to build/deploy
+        # GASNet is as an EMBEDDED library within the build of the client package
+        # (eg. Berkeley UPC, UPC++, Legion, etc), some of which provide build-time
+        # selection of the GASNet library sources. This spack package provides
+        # the GASNet-EX sources, for use by appropriate client packages.
+        install_tree('.', prefix + "/src")
+
+        # Library build is provided for unit-testing purposes only (see notice above)
+        if len(self.conduits()) > 0:
```

Comment:
Correct, that's intentional. 

This is exactly the capability desired by production GASNet client packages (such as UPC++), that can use this feature to configure and build their embedded GASNet-EX libraries from these source files using a Spack-directed version. For example, this PR enables one to easily build UPC++ with any selected GASNet-EX version, via a command like:

`spack install upcxx+gasnet^gasnet@develop`

I realize this might not be exactly common Spack practice, but it enables the intended capability which is desirable for debugging/support scenarios (in our admittedly unusual build paradigm).